crates/worker/src/partition/snapshots/repository.rs

```diff
@@ -306,11 +322,133 @@ impl SnapshotRepository {
 
         Ok(())
     }
-}
 
-/// S3 and other stores require a certain minimum size for the parts of a multipart upload. It is an
-/// API error to attempt a multipart put below this size, apart from the final segment.
-const MULTIPART_UPLOAD_CHUNK_SIZE_BYTES: usize = 5 * 1024 * 1024;
+    /// Discover and download the latest snapshot available. Dropping the returned
+    /// `LocalPartitionSnapshot` will delete the local snapshot data files.
+    #[instrument(
+        level = "debug",
+        skip_all,
+        err,
+        fields(%partition_id),
+    )]
+    pub(crate) async fn get_latest(
+        &self,
+        partition_id: PartitionId,
+    ) -> anyhow::Result<Option<LocalPartitionSnapshot>> {
+        let latest_path = object_store::path::Path::from(format!(
+            "{prefix}{partition_id}/latest.json",
+            prefix = self.prefix,
+            partition_id = partition_id,
+        ));
+
+        let latest = self.object_store.get(&latest_path).await;
+
+        let latest = match latest {
+            Ok(result) => result,
+            Err(object_store::Error::NotFound { .. }) => {
+                debug!("Latest snapshot data not found in repository");
+                return Ok(None);
+            }
+            Err(e) => return Err(e.into()),
+        };
+
+        let latest: LatestSnapshot =
+            serde_json::from_slice(latest.bytes().await?.iter().as_slice())?;
+        trace!("Latest snapshot metadata: {:?}", latest);
+
+        let snapshot_metadata_path = object_store::path::Path::from(format!(
+            "{prefix}{partition_id}/{path}/metadata.json",
```

Comment:
Are you gonna unify these paths once #2310 gets merged?

Reply:
I didn't get to this unfortunately - let me track this separately as https://github.com/restatedev/restate/issues/2389. I don't want to do any last minute fixes right now for fear of breaking things, and it would be great to merge this into main so it doesn't rot sitting in PR for another week 😅